### ISSUE TITLE
Suggestion: Reference Documents

### ISSUE THREAD

**Issue by gubler** [xwmx/nb#54]
I am not sure this is even within the scope of this project and could add unwanted complexity. Or even be worth the time it would take to implement :)

I have several notes that refer to secondary files (such as screenshots, etc.) that would not be notes in themselves. It would be nice if there was a way to add a "reference document/file" that is not a note (and would not pollute `nb ls` or `nb q`) but that I could still link to from my notes and would be synced.

I can think of 2 ways to handle this:

1. Have some way to "flag" a file as a reference document
2. Add a whole new type of "reference document" object

The former would probably need:

- a way to flag/un-flag a file as a reference document
- list all flagged documents
- do all the normal `nb` actions on flagged documents

The latter would probably need the following functionality:

- Adding documents
- Removing document
- Edit document
- Replacing an existing document (new version of file) - this is different from editing a reference document
    - I guess this would be a shortcut for delete/add with the same filename
- Look up all "actual" notes that link to the reference document

**Comment by RogerTouchCR** [comment]
This would be an interesting idea/feature.

I wonder if something like tagging could work, where some markdown syntax could be evaluated? Just a thought.

**Comment by gubler** [comment]
Tagging would work for text documents, but I'm not sure how to tag images/PDF/etc. From what I can see in the documentation, tags have to be text body.

**Comment by 100ideas** [comment]
Also wondering about this, and #42. Use case:

I  added a bookmark for this github project `https://github.com/ottomatica/docable-notebooks` and felt a warm glow that pandoc and readability would convert the readme into a nice markdown doc for me.

The readme links to a related pdf file, which I then wanted to incorporate or link to my bookmark. So I tried 
```
$ bookmark https://github.com/ottomatica/docable-notebooks --title "docable - Next-Generation Interactive Notebooks (github.com)" --tags notebook --related http://chrisparnin.me/pdf/docable_FSE_20.pdf
```
I was hoping the `--related`` argument would actually use `nb import download <url>` under the hood to download the related resource (pdf) since it wasn't a convertable html or md file. Instead I see it just added the url specified to the `## Related` section of the `bookmark.md`. 

I was hoping `nb` would automagically scan the elements of the `## Related` list and show me notes/files/other entries related to the bookmark in the list and search views, the way tags seem to automagically work.

Two other ideas for modeling "related" relationships between nb items:

1. nb elements that relate to a note or bookmark (i.e. would be listed in `## Related` could be indicated on the same line as the notes filename in the `.index` file
```
20201031024333.bookmark.md
20201031024531.bookmark.md
20201031033622.bookmark.md
20201031043634.bookmark.md > docable_FSE_20.pdf (,[etc.jpg], [other-related-note.md])
docable_FSE_20.pdf
```

2. subfolders with same filename as note/bookmark hold arbitrary nb entries that are "related". Don't like this as much since it seems worse for keeping filenames in subfolder in-sync with names referenced in parent note <a> and <img> tags.
```
torchy:~/.n/home on master
$ tree .
.
├── 20201031024333.bookmark.md
├── 20201031024531.bookmark.md
├── 20201031033622.bookmark.md
├── 20201031043634.bookmark.md
            └── 20201031043634_related
                       ├── other-related-note.md
                       └── docable_FSE_20.pdf
```

just some food for thought.

Boostnote does something similar to 2, but generates a UUID for each note's filename, then keeps a single `attachments` folder next to the `notes` folder, with subfolders named for each UUID in `attachments` to hold any resources linked to in the note's markdown.

**Comment by xwmx** [comment]
Yeah, I have similar thoughts along these lines and am whittling in this direction.

**Comment by koss** [comment]
This is what's keeping me from adopting nb. Nearly all notes have associated files. Usually a single pdf.

**Comment by enobrev** [comment]
Big fan of this idea.  By default `typora` does this by adding a ${filename}.assets directory and references the files in that directory.  The settings can be changed to put the files in the current or custom directory as well.

![Screenshot-20201111100502-582x149](https://user-images.githubusercontent.com/584126/98834920-6271e580-2405-11eb-894d-c030b1a7d6b2.png)

[Here's](http://support.typora.io/Images/) what that "Learn More" link leads to

_(I have nothing to do with `typora` besides being a daily user)_
